### PR TITLE
Add ROADMAP document

### PR DESCRIPTION
**Pull request by steveeJ**
Contributes to #125.

/cc @squaremo

### CODE REVIEW

File: ROADMAP.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# CNI Roadmap
+
+This document defines a high level roadmap for CNI development.
+The list below is not complete, and we advise to get the current project state from the [milestones defined in GitHub](https://github.com/containernetworking/cni/milestones).
+
+## CNI Milestones
+
+### [v0.3.0](https://github.com/containernetworking/cni/milestones/v0.3.0)
```
Comment (jonboulle) [inline ROADMAP.md]: peanut gallery, this seems a bit uselessly vague. how about an estimated timeline and/or at least some high level goals for the milestone releases?

Comment (steveeJ) [inline ROADMAP.md]: Yep, would be good to have some real content ;-) I was wondering if we shall do a mix of roadmap and changelog, opinions?

Comment (jonboulle) [inline ROADMAP.md]: I think there's enough in the milestones to extract a few highlights to put here

File: ROADMAP.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+* Add firewalld plugin
+
+### [v1.0.0](https://github.com/containernetworking/cni/milestones/v1.0.0)
+
+- Plugin composition functionality
+- IPv6 support
+- Stable SPEC
+- Strategy and tooling for backwards compatibility
+- Complete test coverage
+- Integrate build artefact generation with CI
```
Comment (fnordahl) [inline ROADMAP.md]: s/artefact/artifact

Comment (jonboulle) [inline ROADMAP.md]: http://grammarist.com/spelling/artefact-artifact/

**Comment by jonboulle** [comment]
lgtm for now

File: ROADMAP.md (hunk 2 above)
Comment (fnordahl) [inline ROADMAP.md]: Ah, all good then. We learn American English here in Norway. Sorry about the noise. O:-)

**Comment by tomdee** [comment]
I'm happy for this to be merged but can we agree that this roadmap hasn't had much discussion (that I'm aware of) so we should be afraid to change it future? I worry that once it's committed we're somehow committed to it.

**Comment by jonboulle** [comment]
I for one certainly wouldn't consider it set in stone

**Comment by steveeJ** [comment]
> I'm happy for this to be merged but can we agree that this roadmap hasn't had much discussion (that I'm aware of) so we should be afraid to change it future? I worry that once it's committed we're somehow committed to it.

It hasn't had much discussion. I assembled it from different conversations, including the recent meetings but also OOB discussions, mostly @squaremo. It's a first impression and I'm certainly not saying this is how it has to be. As always, it's a start.

File: ROADMAP.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+## CNI Milestones
+
+### [v0.2.0](https://github.com/containernetworking/cni/milestones/v0.2.0)
+
+* Signed release binaries
+* Introduction of a testing strategy/framework
+
+### [v0.3.0](https://github.com/containernetworking/cni/milestones/v0.3.0)
+
+* Further Increase test coverage
```
Comment (squaremo) [inline ROADMAP.md]: Nitpick: I don't think "Increase" needs a capital letter here.

**Comment by squaremo** [comment]
I sympathise with Tom's concern -- an explicit roadmap represents a commitment (whatever one says in the front matter). However we do need to balance that with taking _some_ responsibility for making progress.

Perhaps we can compile the roadmap this way:

 * start with what happened in previous releases
 * include proposed (and ideally, discussed) headline issues for one minor version ahead
 * end with aspirations for the next major release

These are all present already -- nice! The difference is that 0.4 and 0.5 would be elided (but may be discoverable from github milestones, which are understood to be provisional). WDYT @steveeJ @tomdee ?

**Comment by steveeJ** [comment]
> The difference is that 0.4 and 0.5 would be elided

I have been trying to keep issues/PRs to an undefined but limited count to allow a more frequent release cycle, represented in the ROADMAP as well as the issue management. 
With your suggestion, would we put these planned changes to the 1.0 milestone? Once released, then create the previous milestones and move those items back?

File: ROADMAP.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+### [v0.3.0](https://github.com/containernetworking/cni/milestones/v0.3.0)
+
+* Further increase test coverage
+* Simpler default route handling in bridge plugin
+* Clarify project description, documentation and contribution guidelines
+
+### [v0.4.0](https://github.com/containernetworking/cni/milestones/v0.4.0)
+
+* Further Increase test coverage
```
Comment (squaremo) [inline ROADMAP.md]: No capital needed for "increase"

**Comment by squaremo** [comment]
Thanks @steveeJ and commenters!

**Pull request merged**